### PR TITLE
Update IntelliJ gradle plugin to 2.0

### PR DESCRIPTION
**Pull request by brian-mcnamara**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -114,12 +112,12 @@ tasks {
...
-        channels = properties("pluginVersion").map { listOf(it.split('-').getOrElse(1) { "default" }.split('.').first()) }
-    }
+//    publishPlugin {
+//        dependsOn("patchChangelog")
+//        token = environment("PUBLISH_TOKEN")
+//        // The pluginVersion is based on the SemVer (https://semver.org) and supports pre-release labels, like 2.1.7-alpha.3
+//        // Specify pre-release label to publish the plugin in a custom Release Channel automatically. Read more:
+//        // https://plugins.jetbrains.com/docs/intellij/deployment.html#specifying-a-release-channel
+//        channels = properties("pluginVersion").map { listOf(it.split('-').getOrElse(1) { "default" }.split('.').first()) }
+//    }
```
Comment (AdwaitChavan) [inline build.gradle.kts:122]: Is this block supposed to be removed?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,119 +1,81 @@
 # GitHub Actions Workflow is created for testing and preparing the plugin release in the following steps:
-# - validate Gradle Wrapper,
-# - run 'test' and 'verifyPlugin' tasks,
-# - run 'buildPlugin' task and prepare artifact for the further tests,
-# - run 'runPluginVerifier' task,
-# - create a draft release.
+# - Validate Gradle Wrapper.
+# - Run 'test' and 'verifyPlugin' tasks.
```
Comment (AdwaitChavan) [inline .github/workflows/build.yml:3]: Are these formatter related changes? Might want to standardize it using `.idea/codeStyles`

File: build.gradle.kts (hunk 1 above)
Comment (brian-mcnamara) [reply build.gradle.kts:122]: Yes, the plugin is not currently published to the marketplace

File: .github/workflows/build.yml (hunk 2 above)
Comment (brian-mcnamara) [reply .github/workflows/build.yml:3]: Nope, updated the file based on the IntelliJ plugin template repo
